server/sonar-server/src/main/java/org/sonar/server/computation/task/projectanalysis/period/Period.java

```diff
@@ -49,13 +52,23 @@ public Period(int index, String mode, @Nullable String modeParameter,
     this.analysisUuid = analysisUuid;
   }
 
+  public Period(String mode, @Nullable String modeParameter, long snapshotDate, String analysisUuid) {
+    this.index = 1;
+    this.mode = requireNonNull(mode);
+    this.modeParameter = modeParameter;
+    this.snapshotDate = snapshotDate;
+    this.analysisUuid = analysisUuid;
+  }
+
   public static boolean isValidPeriodIndex(int i) {
     return i > 0 && i < 6;
   }
 
   /**
    * Index of periods is 1-based. It goes from 1 to 5.
+   * @deprecated only leak period can now be defined
    */
+  @Deprecated
   public int getIndex() {
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO') Do not forget to remove this deprecated code someday. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)